Add tracking of flush strategies.

toregge opened the pull request:
@vekterli : please review

toregge committed:
Add tracking of flush strategies.

toregge committed:
Rename variables to clarify semantics.

vekterli commented on searchcore/src/vespa/searchcore/proton/flushengine/flushengine.cpp:
```diff
@@ -274,10 +286,79 @@ FlushEngine::prune()
+}
+
+void
+FlushEngine::prune_flushing_strategies(std::vector<uint32_t> strategy_ids_for_finished_flushes)
```
Consider passing by const ref or `std::move`-ing at the callsites, since these seem to currently pass by value

vekterli reviewed (dismissed):
:+1:

toregge committed:
Use std::move to move data instead of copying it.
Add comment describing semantics of _flushing_strategies member variable.
Assert that flushing task is found in flushDone member function.
